emailaddress/client_test.go

```diff
@@ -0,0 +1,149 @@
+package emailaddress
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+	"github.com/clerk/clerk-sdk-go/v2/clerktest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEmailAddressClientCreate(t *testing.T) {
+	t.Parallel()
+	email := "foo@bar.com"
+	userID := "user_123"
+	id := "idn_123"
+	config := &ClientConfig{}
+	config.HTTPClient = &http.Client{
+		Transport: &clerktest.RoundTripper{
+			T:      t,
+			In:     json.RawMessage(fmt.Sprintf(`{"email_address":"%s","user_id":"%s","verified":false}`, email, userID)),
+			Out:    json.RawMessage(fmt.Sprintf(`{"id":"%s","email_address":"%s"}`, id, email)),
+			Method: http.MethodPost,
+			Path:   "/v1/email_addresses",
+		},
+	}
+	client := NewClient(config)
+	emailAddress, err := client.Create(context.Background(), &CreateParams{
+		UserID:       clerk.String(userID),
+		EmailAddress: clerk.String(email),
+		Verified:     clerk.Bool(false),
+	})
+	require.NoError(t, err)
+	require.Equal(t, id, emailAddress.ID)
+	require.Equal(t, email, emailAddress.EmailAddress)
+}
+
+func TestEmailAddressClientCreate_Error(t *testing.T) {
+	t.Parallel()
+	config := &ClientConfig{}
+	config.HTTPClient = &http.Client{
+		Transport: &clerktest.RoundTripper{
+			T:      t,
+			Status: http.StatusBadRequest,
+			Out: json.RawMessage(`{
+  "errors":[{
+		"code":"create-error-code"
+	}],
+	"clerk_trace_id":"create-trace-id"
+}`),
+		},
+	}
+	client := NewClient(config)
+	_, err := client.Create(context.Background(), &CreateParams{})
+	require.Error(t, err)
+	apiErr, ok := err.(*clerk.APIErrorResponse)
```

Comment:
💡 Perhaps a helper is useful?

```go
apiErr, ok := clerk.AsErrorResponse(err)
```

Reply:
Good idea. However,  I cannot see how this would help. 

I imagine that if we change the `APIErrorResponse` type, even if we had the helper, we would still have to change the code that uses the error response type. :thinking: 

Happy to add the helper in a separate PR.

Reply:
It would mostly serve as a guide. Autocompletion may help you write the conversion faster too. Otherwise someone must probably remember or look up the error type? The rest of the code will remain the same, you're absolutely right!